### PR TITLE
GH-5543 - cherry pick from release/1.3.x to develop

### PR DESCRIPTION
**Pull request by plabadie**
closes GH-5543

(cherry picked from commit 1c15ccd39cab966dc487e4456e1a90687768e0c2)

### CODE REVIEW

File: projects/core/src/user/facade/user-address.service.ts [hunk 1]
```diff
@@ -87,38 +78,30 @@ export class UserAddressService {
...
-            addressId: addressId,
-            address: address,
-          })
-        )
+    this.withUserId(userId =>
+      this.store.dispatch(
+        new UserActions.UpdateUserAddress({
+          userId,
+          addressId,
+          address: address,
```
Comment (tobi-or-not-tobi) [inline projects/core/src/user/facade/user-address.service.ts]: you can leave out the property name here

File: projects/core/src/user/facade/user-consent.service.ts [hunk 2]
```diff
@@ -11,6 +11,7 @@ import {
 } from 'rxjs/operators';
 import { AuthService } from '../../auth/facade/auth.service';
 import { Consent, ConsentTemplate } from '../../model/consent.model';
+import { OCC_USER_ID_CURRENT } from '../../occ';
```
Comment (tobi-or-not-tobi) [inline projects/core/src/user/facade/user-consent.service.ts]: not sure, but often this is causing issues with ngpackagr build. No need to load from the barrel file?

File: projects/core/src/user/facade/user-consent.service.ts [hunk 3]
```diff
@@ -224,18 +219,14 @@ export class UserConsentService {
...
-        this.store.dispatch(
-          new UserActions.WithdrawUserConsent({
-            userId: occUserId,
-            consentCode,
-          })
-        )
+    this.withUserId(userId =>
+      this.store.dispatch(
+        new UserActions.WithdrawUserConsent({
+          userId: userId,
```
Comment (tobi-or-not-tobi) [inline projects/core/src/user/facade/user-consent.service.ts]: we can leave out the property

File: projects/core/src/user/facade/user-payment.service.ts [hunk 4]
```diff
@@ -67,18 +66,14 @@ export class UserPaymentService {
...
-        this.store.dispatch(
-          new UserActions.SetDefaultUserPaymentMethod({
-            userId: occUserId,
-            paymentMethodId,
-          })
-        )
+    this.withUserId(userId =>
+      this.store.dispatch(
+        new UserActions.SetDefaultUserPaymentMethod({
+          userId: userId,
```
Comment (tobi-or-not-tobi) [inline projects/core/src/user/facade/user-payment.service.ts]: property can be removed

Comment (tobi-or-not-tobi) [reply projects/core/src/user/facade/user-payment.service.ts]: property key I mean

File: projects/core/src/user/facade/user-payment.service.ts [hunk 5]
```diff
@@ -87,18 +82,14 @@ export class UserPaymentService {
...
-        this.store.dispatch(
-          new UserActions.DeleteUserPaymentMethod({
-            userId: occUserId,
-            paymentMethodId,
-          })
-        )
+    this.withUserId(userId =>
+      this.store.dispatch(
+        new UserActions.DeleteUserPaymentMethod({
+          userId: userId,
```
Comment (tobi-or-not-tobi) [inline projects/core/src/user/facade/user-payment.service.ts]: same

**Comment by plabadie** [comment]
> LGTM
> 
> Only the barrel import I worry about, but that might have been the same on the 1.3.x branch earlier today.

Yes it is the same on the 1.3.x and 1.3.1 branches: this PR is the same commit copied by `git cherry-pick`

Therefore, 
1) I fixed the imports in this PR to develop.
2) I pushed 2 PRs ( #5659 , #5660  )to fix the imports in 1.3.x and 1.3.1 linked to ticket GH-5658.

File: projects/core/src/user/facade/user-consent.service.ts (hunk 2 above)
Comment (plabadie) [reply projects/core/src/user/facade/user-consent.service.ts]: Good catch overall. 
I agree this has cause issues in the past and the fix is so small it's not worth taking the risk.  See this comment for details on the fix:  https://github.com/SAP/cloud-commerce-spartacus-storefront/pull/5654#issuecomment-561870734

File: projects/core/src/user/facade/user-address.service.ts (hunk 1 above)
Comment (plabadie) [reply projects/core/src/user/facade/user-address.service.ts]: Fixed.

File: projects/core/src/user/facade/user-consent.service.ts (hunk 3 above)
Comment (plabadie) [reply projects/core/src/user/facade/user-consent.service.ts]: Fixed

File: projects/core/src/user/facade/user-payment.service.ts (hunk 4 above)
Comment (plabadie) [reply projects/core/src/user/facade/user-payment.service.ts]: Fixed

File: projects/core/src/user/facade/user-payment.service.ts (hunk 5 above)
Comment (plabadie) [reply projects/core/src/user/facade/user-payment.service.ts]: Fixed

**Pull request merged**